src/python/pants/backend/python/tasks/python_binary_create.py

```diff
@@ -77,6 +87,17 @@ def __init__(self, *args, **kwargs):
     super().__init__(*args, **kwargs)
     self._distdir = self.get_options().pants_distdir
 
+  @property
+  def _generate_ipex(self) -> bool:
+    return cast(bool, self.get_options().generate_ipex)
```

Comment:
Nit: Would this be an overkill for `self.get_options().generate_ipex`, or is the goal to get it type annotation? If the latter, I can imagine the type annotation framework should support this at the option module level instead of an adhoc implementation here.

Reply:
The goal was just to make this type-annotated (otherwise mypy complain), and this is a construction I've added to other places with a TODO to make pants options eventually respect types. I absolutely think that it's possible to make pants options "know" the types automatically -- it would require some extra thought that I haven't yet put into this yet. Possibly relevant is #7567?

Reply:
Opened #9280 on this topic!!